packages/flutter/lib/src/material/tabs.dart

```diff
@@ -586,57 +596,37 @@ class _IndicatorPainter extends CustomPainter {
 
     final double index = controller.index.toDouble();
     final double value = controller.animation!.value;
-
-    // The progress of the animation from 0 to 1.
-    late double tabChangeProgress;
-
-    // If we are changing tabs via index, we want to map the progress between 0 and 1.
-    if (controller.indexIsChanging) {
-      double progressLeft = (index - value).abs();
-      final int tabsDelta = (controller.index - controller.previousIndex).abs();
-      if (tabsDelta != 0) {
-        progressLeft /= tabsDelta;
-      }
-      tabChangeProgress = 1 - clampDouble(progressLeft, 0.0, 1.0);
-    } else {
-      // Otherwise, the progress is how close we are to the current tab.
-      tabChangeProgress = (index - value).abs();
-    }
+    final double tabChangeProgress = (index - value).abs();
 
     // If the animation has finished, there is no need to apply the stretch effect.
     if (tabChangeProgress == 1.0) {
       return rect;
     }
 
-    // The maximum amount of extra width to add to the indicator.
-    final double stretchSize = rect.width;
+    final bool isMovingRight = rect.left < targetRect.left;
+    final double leftFraction;
+    final double rightFraction;
 
-    final double inflationPerSide = stretchSize * _stretchAnimation.transform(tabChangeProgress) / 2;
-    final Rect stretchedRect = _inflateRectHorizontally(rect, inflationPerSide);
+    if (isMovingRight) {
+      leftFraction = accelerateInterpolation(tabChangeProgress);
+      rightFraction = accelerateInterpolation(tabChangeProgress);
+    } else {
+      leftFraction = decelerateInterpolation(tabChangeProgress);
+      rightFraction = decelerateInterpolation(tabChangeProgress);
```

Comment:
If these are just the same values for right and left in both cases, why not use one variable?

Reply:
I also thought of using a switch statement + pattern combo for assignment, but I figure that implementation preference is not really important.  🙂